privacy-contracts/src/main/solidity/OnChainPrivacyGroupManagementProxy.sol

```diff
@@ -10,28 +11,25 @@ contract OnChainPrivacyGroupManagementProxy is OnChainPrivacyGroupManagementInte
         implementation = _implementation;
     }
 
-    function upgradeTo(address _newImplementation) external {
-        require(implementation != _newImplementation);
-        _setImplementation(_newImplementation);
-    }
-
     function _setImplementation(address _newImp) internal {
         implementation = _newImp;
     }
 
-    function addParticipants(bytes32 enclaveKey, bytes32[] memory participants) public returns (bool) {
+    function addParticipants(bytes32[] memory _publicEnclaveKeys) public returns (bool) {
         OnChainPrivacyGroupManagementInterface privacyInterface = OnChainPrivacyGroupManagementInterface(implementation);
-        return privacyInterface.addParticipants(enclaveKey, participants);
+        return privacyInterface.addParticipants(_publicEnclaveKeys);
     }
 
-    function getParticipants(bytes32 enclaveKey) view public returns (bytes32[] memory) {
+    function getParticipants() view public returns (bytes32[] memory) {
         OnChainPrivacyGroupManagementInterface privacyInterface = OnChainPrivacyGroupManagementInterface(implementation);
-        return privacyInterface.getParticipants(enclaveKey);
+        return privacyInterface.getParticipants();
     }
 
-    function removeParticipant(bytes32 enclaveKey, bytes32 account) public returns (bool) {
+    function removeParticipant(bytes32 _member) public returns (bool) {
         OnChainPrivacyGroupManagementInterface privacyInterface = OnChainPrivacyGroupManagementInterface(implementation);
-        return privacyInterface.removeParticipant(enclaveKey, account);
+        bool result = privacyInterface.removeParticipant(_member);
+        emit ParticipantRemoved(result, _member);
```

Comment:
Instead of always emitting the event with a status, why don't we only emit the event when the participant is removed successfully?

Reply:
Good plan!